code/RealMeAuthenticator.php

```diff
@@ -25,51 +25,26 @@ protected static function on_register()
         $cache = SS_Cache::factory('RealMeAuthenticator');
 
         $cacheKey = 'RegisterCheck';
-        if (true === (bool) $cache->load($cacheKey)) {
+        if ((bool) $cache->load($cacheKey)) {
```

Comment:
You can use `$cache->test($cacheKey)` to check a cache exists - probably better than loading the whole cached data just to convert to a bool